src/hash.h

```diff
@@ -223,15 +221,6 @@ class HashedSourceWriter : public HashWriter
     }
 };
 
-/** Compute the 256-bit hash of an object's serialization. */
-template<typename T>
-uint256 SerializeHash(const T& obj, int nType=SER_GETHASH, int nVersion=PROTOCOL_VERSION)
```

Comment:
Why get rid of this helper function? `SerializeHash(obj)` seems simpler to read than `(HashWriter{} << obj).GetHash()`.

Reply:
I guess most of the code is combining a few objects and hashing them, so I guess you're right...

Reply:
```c++
    template <typename... Ts>
    explicit HashWriter(const Ts&... obj)
    {
        (*this << ... << obj);
    }
```

lets you replace

```c++
    uint64_t hash1 = (HashWriter{} << nKey << GetKey()).GetCheapHash();
    uint64_t hash2 = (HashWriter{} << nKey << netgroupman.GetGroup(*this) << (hash1 % ADDRMAN_TRIED_BUCKETS_PER_GROUP)).GetCheapHash();
```

with

```c++
    uint64_t hash1 = HashWriter{nKey, GetKey()}.GetCheapHash();
    uint64_t hash2 = HashWriter{nKey, netgroupman.GetGroup(*this), (hash1 % ADDRMAN_TRIED_BUCKETS_PER_GROUP)}.GetCheapHash();
```

Not sure it's worth it though.